### PR TITLE
Fix use nil error

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Fix use nil error

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json_test.go [hunk 1]
```diff
@@ -398,7 +398,7 @@ func TestDecode(t *testing.T) {
 
 		switch {
 		case err == nil && test.errFn != nil:
-			t.Errorf("%d: failed: %v", i, err)
+			t.Errorf("%d: failed: %v", i, "not getting the expected error")
```
Comment (apelisse) [inline staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json_test.go:401]:
```suggestion
			t.Errorf("%d: failed: not getting the expected error", i)
```

Comment (mengjiao-liu) [reply staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json_test.go:401]: Thanks for your advice. I'll fix it

**Comment by pacoxu** [comment]
/test pull-kubernetes-node-e2e-containerd
for CI hang

**Pull request merged**